packages/smooth_app/test/plural_translation_test.dart

```diff
@@ -8,62 +8,231 @@ void main() {
+          //Check if any translation contains numbers
+          expect(minutes.any((String x) => x.contains(RegExp(r'[0-9]'))), true);
+          expect(hours.any((String x) => x.contains(RegExp(r'[0-9]'))), true);
+          expect(days.any((String x) => x.contains(RegExp(r'[0-9]'))), true);
+          expect(weeks.any((String x) => x.contains(RegExp(r'[0-9]'))), true);
+          expect(months.any((String x) => x.contains(RegExp(r'[0-9]'))), true);
+          expect(compare.any((String x) => x.contains(RegExp(r'[0-9]'))), true);
 
-            final List<String> minutes = <String>[];
-            final List<String> hours = <String>[];
-            final List<String> days = <String>[];
-            final List<String> weeks = <String>[];
-            final List<String> months = <String>[];
-            final List<String> compare = <String>[];
+          const String crazyString = 'の中ழ்';
+          const Object crazyObject = crazyString;
 
-            for (int i = -1; i < 1001; i++) {
-              minutes.add(appLocalizations!.plural_ago_minutes(i));
-              hours.add(appLocalizations!.plural_ago_hours(i));
-              days.add(appLocalizations!.plural_ago_days(i));
-              weeks.add(appLocalizations!.plural_ago_weeks(i));
-              months.add(appLocalizations!.plural_ago_months(i));
-              compare.add(appLocalizations!.plural_compare_x_products(i));
-            }
+          /// int value designed to trigger the "OTHER" case in plural labels.
+          ///
+          /// That's not that easy for some languages,
+          /// cf. https://github.com/dart-lang/intl/blob/master/lib/src/plural_rules.dart
+          /// The value should
+          /// * end with a 4, 6 or 9 for 'fil' and 'tl',
+          ///   cf. PluralCase _fil_rule()
+          /// * not end with 0, 1, 11-19 for 'lv',
+          ///   cf. PluralCase _lv_rule()
+          const int crazyInt = 2080706059;
 
-            //Check if any translation contains numbers
-            expect(
-                minutes.any((String x) => x.contains(RegExp(r'[0-9]'))), true);
-            expect(hours.any((String x) => x.contains(RegExp(r'[0-9]'))), true);
-            expect(days.any((String x) => x.contains(RegExp(r'[0-9]'))), true);
-            expect(weeks.any((String x) => x.contains(RegExp(r'[0-9]'))), true);
-            expect(
-                months.any((String x) => x.contains(RegExp(r'[0-9]'))), true);
-            expect(
-                compare.any((String x) => x.contains(RegExp(r'[0-9]'))), true);
-          });
-          break;
+          expect(
+            appLocalizations.sign_up_page_username_length_invalid(crazyInt),
+            contains(crazyInt.toString()),
+          );
+          expect(
+            appLocalizations.pct_match(crazyObject),
+            contains(crazyObject.toString()),
+          );
+          expect(
+            appLocalizations.contact_form_body_android(
+                crazyInt, '', '', '', '', ''),
```

Comment:
This and the following with empfehlen strings doesn't necessarily assure that all of the variables are still valid in all languages. We should check if crazyInt is in it 6 times if there is a check for it

Reply:
@M123-dev I don't agree with you: I test if each parameter is displayed. Good enough is good enough.
What you suggest would mean that in some cases, the translation tool mistakes a parameter for another, which is very unlikely.
Anyway, that test set is a start, and is open to evolution.